hyperopt/tests/test_atpe_basic.py

```diff
@@ -25,7 +25,8 @@ def objective(args):
         # minimize the objective over the space
         from hyperopt import fmin, atpe, space_eval
 
-        best = fmin(objective, space, algo=atpe.suggest, max_evals=10)
+        # Max evals should be greater than 10, as the first 10 runs are only the initialization rounds
+        best = fmin(objective, space, algo=atpe.suggest, max_evals=20)
```

Comment:
The first thing i thought when I saw your change was: 

"maybe rather than increasing the max_evals to 20, you can reduce the init rounds to e.g. 2-3?" 

however, you're editing another test, which is supposed to test something. I know it's not explicitly part of your task, but could you turn the 

```
print(best)
        # -> {'a': 1, 'c2': 0.01420615366247227}
        print(space_eval(space, best))
        # -> ('case 2', 0.01420615366247227}
```
into actual checks with unittest assert? and please move imports to the top and run isort, black

I hope this shouldn't take you more than 10min and would increase the hygiene of this file. I would also ask you to turn the test into pytest but appreciate that's too much to ask. If you need a hand, please let me know!

Reply:
Updated the test. I had to change the objective function so that its more consistently returns case 2.

Reply:
some comments:

- what do you mean it's more consistently ...? it's not ALWAYS returning case2? if so, please set the random seed to ensure determinism
- could you reduce the initial rounds to e.g. 5, so max_evals can stay the same?
- you can remove the class as well and just leave `test_run_basic_search` at the top level

Reply:
- Yes, it was not always returning case2. But if we are only checking if end2end works then I guess this can be ignored and I'll revert it.

- I can't really change the initial rounds in a test case without altering the code for ATPE, which I think would be better left avoided. See https://github.com/hyperopt/hyperopt/blob/8edf67974e68a01ed553f768166bb83812ac0578/hyperopt/atpe.py#L715

- Alright will do

Reply:
Your approach is perfect with those asserts. I just meant to have something that's deterministic, even if practically speaking is not a fancy objective function doing something realistic. Forget about changing the objective function, just set the random seed so that it always returns the same, and it's good to go! Sorry for the confusion

Reply:
Ok thanks for clearing that up. I'm still having some issues setting the random state. I'm setting `rstate=np.random.RandomState(1)` however the results still change. I tried the same with `tpe.suggest` and it seems to work fine. Am I missing something here?

Reply:
https://stackoverflow.com/a/22994890 ?

Reply:
might also be that the `random` python library is used, so you need to fix that seed as well? This is definitely something to review, we should set the global random seeds before running tests, not for each test separately (but not part of your PR ofc)

Reply:
yeah that seems to be the case. I Needed to set both `random.seed` and `numpy.random.seed`. 

https://github.com/hyperopt/hyperopt/blob/8edf67974e68a01ed553f768166bb83812ac0578/hyperopt/atpe.py#L698